sdk/eventhub/azure-eventhub/tests/livetest/asynctests/test_send_async.py

```diff
@@ -223,3 +225,191 @@ async def test_send_batch_pid_pk_async(invalid_hostname, partition_id, partition
+    assert partition_publishing_properties_2["last_published_sequence_number"] == 1
+    assert partition_publishing_properties_2["producer_group_id"] == 0
+    assert partition_publishing_properties_2["owner_level"] == partition_publishing_properties["owner_level"] + 1
+
+    # test producer-epoch-stolen
+    with pytest.raises(EventDataSendError):
+        # the first producer will get exploded as the link is stolen by the second producer with higher owner level
+        try:
+            await client.send_batch([EventData(payload)], partition_id="0")
+        except EventDataSendError as e:
+            assert e.error == "producer-epoch-stolen"
+            raise e
+
+    await client.close()
+    await client_2.close()
+
+
+@pytest.mark.liveTest
+@pytest.mark.asyncio
+async def test_idempotent_sender_wrong_partition_configs_async(connstr_receivers):
+    connection_str, receivers = connstr_receivers
+    payload = "A1"
+    with pytest.raises(ValueError):
+        PartitionPublishingConfiguration(owner_level=2**15)
+
+    with pytest.raises(ValueError):
+        PartitionPublishingConfiguration(producer_group_id=-1)
+
+    with pytest.raises(ValueError):
+        EventHubProducerClient.from_connection_string(
+            connection_str,
+            enable_idempotent_partitions=True,
+            partition_configs={"0": {"starting_sequence_number": 2**31}}
+        )
+
+    partition_configs = {
+        "0": {
+            "producer_group_id": 100,
+            "owner_level": 100,
+            "starting_sequence_number": 100
+        }
+    }
+    client = EventHubProducerClient.from_connection_string(
+        connection_str,
+        enable_idempotent_partitions=True,
+        partition_configs=partition_configs
+    )
+
+    # test invalid init setting
+    with pytest.raises(ConnectionLostError):
+        try:
+            # The service would return link detach with error code being "not allowed"
+            await client.send_batch([EventData(payload)], partition_id="0")
+            await client.close()
+        except ConnectionLostError as e:
+            assert e.details.condition == uamqp_constants.ErrorCodes.NotAllowed
+            raise e
+
+    client = EventHubProducerClient.from_connection_string(
+        connection_str,
+        enable_idempotent_partitions=True
+    )
+
+    event_data_list = [EventData(payload) for _ in range(10)]
+    await client.send_batch(event_data_list, partition_id="0")
+
+    partition_publishing_properties = await client.get_partition_publishing_properties("0")
+
+    # test out-of-sequence
+    partition_config_for_producer_2 = {
+        "0": {
+            "producer_group_id": partition_publishing_properties["producer_group_id"],
+            "owner_level": partition_publishing_properties["owner_level"],
+            "starting_sequence_number": partition_publishing_properties["last_published_sequence_number"] - 5
+        }
```

Comment:
what happens if you send in an empty dict --> even though you set `enable_idempotent_partitions=True`?

Reply:
the empty dict will simply be ignored, no settings on the producer link